source/acquaman/AMGenericContinuousScanConfiguration.cpp

```diff
@@ -0,0 +1,328 @@
+#include "AMGenericContinuousScanConfiguration.h"
+
+#include "ui/acquaman/AMGenericContinuousScanConfigurationView.h"
+#include "acquaman/AMGenericContinuousScanController.h"
+#include "beamline/AMBeamline.h"
+
+#include <math.h>
+
+AMGenericContinuousScanConfiguration::AMGenericContinuousScanConfiguration(QObject *parent)
+	: AMContinuousScanConfiguration(parent)
+{
+	setName("Generic Continuous Scan");
+	setUserScanName("Generic Continuous Scan");
+}
+
+AMGenericContinuousScanConfiguration::AMGenericContinuousScanConfiguration(const AMGenericContinuousScanConfiguration &original)
+	: AMContinuousScanConfiguration(original)
+{
+	setName(original.name());
+	setUserScanName(original.name());
+	i0_ = original.i0();
+
+	foreach (AMRegionOfInterest *region, original.regionsOfInterest())
+		addRegionOfInterest(region->createCopy());
+
+	computeTotalTime();
+
+	for (int i = 0, size = scanAxes_.count(); i < size; i++){
+
+		connect(scanAxisAt(i)->regionAt(0), SIGNAL(regionStartChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(i)->regionAt(0), SIGNAL(regionStepChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(i)->regionAt(0), SIGNAL(regionEndChanged(AMNumber)), this, SLOT(computeTotalTime()));
+		connect(scanAxisAt(i)->regionAt(0), SIGNAL(regionTimeChanged(AMNumber)), this, SLOT(computeTotalTime()));
+	}
+}
+
+AMGenericContinuousScanConfiguration::~AMGenericContinuousScanConfiguration()
+{
+
+}
+
+AMScanConfiguration *AMGenericContinuousScanConfiguration::createCopy() const
+{
+	AMScanConfiguration *configuration = new AMGenericContinuousScanConfiguration(*this);
+	configuration->dissociateFromDb(true);
+	return configuration;
+}
+
+AMScanController *AMGenericContinuousScanConfiguration::createController()
+{
+	AMScanActionController *controller = new AMGenericContinuousScanController(this);
+	controller->buildScanController();
+
+	return controller;
+}
+
+AMScanConfigurationView *AMGenericContinuousScanConfiguration::createView()
+{
+	return new AMGenericContinuousScanConfigurationView(this, AMBeamline::bl()->exposedControls(), AMBeamline::bl()->exposedDetectors());
+}
+
+QString AMGenericContinuousScanConfiguration::technique() const
+{
+	return "Generic Continuous Scan";
+}
+
+QString AMGenericContinuousScanConfiguration::description() const
+{
+	return "Generic Continuous Scan";
+}
+
+QString AMGenericContinuousScanConfiguration::detailedDescription() const
+{
+	return "Does a generic continuous scan over a control with a variety of detectors.";
+}
+
+bool AMGenericContinuousScanConfiguration::hasI0() const
+{
+	return i0_.name() != "Invalid Detector";
```

Comment:
Is it possible for i0_ to be a 0 pointer? If so, this line could cause a problem.